### ISSUE TITLE
Unrecognized command line arguments

### ISSUE THREAD

**Issue by slochower** [greenelab/manubot#52]
I'm mystified why this is happening, but I tried running `manubot` commit 33e512d21218263423de5f0d127aac4f8635468f with the `meta-review` manuscript repository and I'm getting a `--log-level` error. I've no idea why this is happening and at this point, I'd like to know if this is reproducible for others. I can run the command without `--log-level`.

```
$ manubot process   \
--content-directory=content \
--output-directory=output  \
--template-variables-path=analyses/deep-review-contrib/deep-review-stats.json  \
--cache-directory=ci/cache   \
--log-level=INFO

usage: manubot [-h] [--version]
               [--log-level {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
               {process,cite} ...
manubot: error: unrecognized arguments: --log-level=INFO
```

**Comment by slochower** [comment]
Ah, this is actually explained by the help message. Oops. The following command is successful.
```
$ manubot --log-level=INFO process   --content-directory=content   --output-directory=output   --template-variables-path=analyses/deep-review-contrib/deep-review-stats.json   --cache-directory=ci/cache
```
Do we want to force `--log-level` to be before `process`?

**Comment by dhimmel** [comment]
Yeah it's a bit confusing. It's possible we could change it so --log-level must come after the subcommand. I don't think there is an elegant implementation that allows it to come before or after the subcommand.

@agitter what do you think?

**Comment by agitter** [comment]
That is confusing behavior.  Could we catch errors for arguments to `manubot` (as opposed to the subcommands) to provide a more specific error message?  If we could override the argparse error message and instead say
> manubot: error: --log-level=INFO must come before the subcommand

it could help.

We could also extend the help message to state
> Manubot: the manuscript bot for scholarly writing
The following optional arguments must come before the subcommand

**Comment by dhimmel** [comment]
@agitter I'm starting to prefer just moving --log-level to the subcommands. Do you prefer either of your two documentation improvements over just moving it (which I think will result in a less confusing error message).

**Comment by agitter** [comment]
> Do you prefer either of your two documentation improvements over just moving it

No.

If we anticipate that putting --log-level after the subcommand is what users would expect without studying the help message, we should do that.  The help message could get cumbersome if there are multiple arguments that apply to all subcommands, but that is not an issue yet.

**Comment by dhimmel** [comment]
Okay, this change is on the plate after #49 and #51. Will probably bundle it with rearranging the `--allow-invalid-csl-data` option and moving some of the code in `manubot/cite/__init__.py` to `manubot/cite/cite_command.py`.

**Comment by dhimmel** [comment]
> Will probably bundle it with rearranging the --allow-invalid-csl-data option and moving some of the code in manubot/cite/__init__.py to manubot/cite/cite_command.py.

PR open for review at https://github.com/greenelab/manubot/pull/54. I did not also make `--allow-invalid-csl-data` available for `manubot process` because I don't think we actually want to allow invalid CSL there since it often breaks the downstream pandoc-citeproc.

If I now run:

```
manubot process   \
--content-directory=content \
--output-directory=output  \
--template-variables-path=analyses/deep-review-contrib/deep-review-stats.json  \
--cache-directory=ci/cache   \
--log-levellls=INFO
```

note the typo `--log-levellls`, I get:

```
usage: manubot [-h] [--version] {process,cite} ...
manubot: error: unrecognized arguments: --log-levellls=INFO
```

I wish in this case it would give the info for the subcomand. Will look into that.
